schematics/shell/index.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  Rule,
+  SchematicContext,
+  Tree,
+  chain,
+  noop,
+  SchematicsException
+} from '@angular-devkit/schematics';
+import {Schema} from './schema';
+import {materialVersion, cdkVersion, angularVersion} from '../utils/lib-versions';
+import {getConfig} from '../utils/devkit-utils/config';
+import {addModuleImportToRootModule} from '../utils/ast';
+import {addHeadLink} from '../utils/html';
+import {addPackageToPackageJson} from '../utils/package';
+
+/**
+ * Scaffolds the basics of a Angular Material application, this includes:
+ *  - Add Packages to package.json
+ *  - Adds pre-built themes to styles.ext
+ *  - Adds Browser Animation to app.momdule
+ */
+export default function(options: Schema): Rule {
+  return chain([
+    options && options.skipPackageJson ? noop() : addMaterialToPackageJson(options),
+    addImportToStyles(options),
+    addAnimationRootConfig(),
+    addFontsToIndex()
+  ]);
+}
+
+/**
+ * Add material, cdk, annimations to package.json
+ */
+function addMaterialToPackageJson(options: Schema) {
+  return (host: Tree) => {
+    addPackageToPackageJson(host, 'dependencies', '@angular/cdk', cdkVersion);
+    addPackageToPackageJson(host, 'dependencies', '@angular/material', materialVersion);
+    addPackageToPackageJson(host, 'dependencies', '@angular/animations', angularVersion);
+    return host;
+  };
+}
+
+/**
+ * Add pre-built styles to style.ext file
+ */
+function addImportToStyles(options: Schema) {
+  return (host: Tree) => {
+    const config = getConfig(host);
+    config.apps.forEach(app => {
+      const themeName = options && options.theme ? options.theme : 'indigo-pink';
```

Comment:
What do you think about making it create a custom theme file by default? I believe that most people will want to customize the theme to one of the non-standard options, so it will be one less step when they want to swap it out.

Reply:
Ya, I'll add an option for custom theme :)

Reply:
@jelbourn - Custom theme requires SCSS be enabled. If they don't have this do we want to configure it for them automatically?